framework/src/materials/DiscreteMaterial.C

```diff
@@ -0,0 +1,44 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+// MOOSE includes
+#include "DiscreteMaterial.h"
+
+template<>
+InputParameters validParams<DiscreteMaterial>()
+{
+  InputParameters params = validParams<Material>();
+  params.set<bool>("allow_cyclic_dependency") = true;
```

Comment:
I think you should have a Material BASE CLASS. You'll inherit `Material` and `DiscreteMaterial from that`. Then you won't need to mess with the Resolver at all for `DiscreteMaterial`.